test/test_fx.py

```diff
@@ -1842,7 +1843,140 @@ def forward(self, x: torch.Tensor):
         graph = ast_rewriter.trace(M())
         traced = GraphModule(ast_rewriter.root, graph, "gm")
 
-        traced.graph.lint(traced)
+        traced.graph.lint()
+
+    def test_submodule_manipulation_API(self):
+        class C(torch.nn.Module):
+            def __init__(self):
+                super(C, self).__init__()
+                self.conv = torch.nn.Conv2d(16, 33, 3, stride=2)
+                self.param = torch.nn.Parameter(torch.rand(2, 3))
+                self.register_buffer("buf", torch.randn(2, 3))
+
+            def forward(self, x):
+                return self.conv(torch.cat([self.param, x, self.buf]))
+
+        class B(torch.nn.Module):
+            def __init__(self):
+                super(B, self).__init__()
+                self.linear = torch.nn.Linear(100, 200)
+                self.net_c = C()
+
+            def forward(self, x):
+                return self.linear(self.net_c(x))
+
+        class A(torch.nn.Module):
+            def __init__(self):
+                super(A, self).__init__()
+                self.net_b = B()
+
+            def forward(self, x):
+                return self.net_b(x)
+
+        a = symbolic_trace(A())
+
+        a.add_submodule("net_b.net_c.dropout", torch.nn.Dropout(p=0.2))
+
+        conv = [n for n in a.graph.nodes if n.target == "net_b.net_c.conv"][-1]
+        with a.graph.inserting_before(conv):
+            dropout = a.graph.call_module(module_name="net_b.net_c.dropout",
+                                          args=conv.args)
+
+        conv.replace_all_uses_with(dropout)
+        a.graph.erase_node(conv)
+        a.recompile()
+
+        def module_exists(gm: GraphModule, path: str) -> bool:
+            atoms: List[str] = path.split(".")
+
+            for item in atoms:
+
+                if not hasattr(gm, item):
+                    return False
+
+                gm = getattr(gm, item)
+
+                if not isinstance(gm, torch.nn.Module):
+                    return False
+
+            return True
```

Comment:
The fact that this is a copy-paste of one of the APIs being tested indicates that this probably isn't meaningful. I think it should be sufficient just to test `get_submodule` and omit the `module_exist` checks here.

If you want to check that the new APIs work with the existing infrastructure in `nn.Module`, you probably want to replace `module_exists` with a check that ensures the qualified name appears in `named_modules`, `_modules`, etc